05-html-css/css-grid-intro.md

```diff
@@ -0,0 +1,334 @@
+
+For the following section we will be using [this example on Codepen.](https://codepen.io/adadev/pen/bvbKQX?editors=1100) Follow along, and comment and uncomment sections as we go through.
+
+#### Grid Template Columns, Grid Template Rows
+
+To define how many columns there are and what size each column is, you use the CSS attribute `grid-template-columns` so that _it applies to your grid container_. For every column, you provide a value for width, separated by **spaces, with no punctuation**.
+
+```css
+.container {
+  display: grid;
+  grid-template-columns: 200px 100px 50%;
+}
+```
+
+How many columns are there? Why are there two rows?
+
+Let's define rows now. To define how many rows there are and what size each row is, we use `grid-template-rows`.
+
+```css
+.container {
+  display: grid;
+  grid-template-columns: 200px 100px 50%;
+  grid-template-rows: 100px 500px;
+}
+```
+
+How many rows are there? **Experiment:** What happens if you add more grid items **and** remove the rule `min-height` defined at the bottom of the Codepen? What is the height of all following rows?
+
+**Practice:** Take a minute to play around. Add more columns and rows to these properties. Remove some.
+
+#### A shortcut: Grid Template
+
+It is common to need to define both the `grid-template-rows` and `grid-template-columns` at the same time. There is a syntax shortcut for this: instead of the above properties, use `grid-template`. The value for this is what you would put in `grid-template-rows`, a slash, then what you would put in `grid-template-columns`.
+
+```css
+.container {
+  display: grid;
+  grid-template: 100px 500px / 200px 100px 50%;
+}
+```
+
+This one line is identical to the two properties we saw above.
+
+### The `fr` fractional unit
+
+CSS Grid lets us use one specific unit of measurement that we can't use anywhere else in CSS: the `fr` fractional unit.
+
+Fractional units are relative units: After all other kinds of units are measured, fractional units take the remaining space and calculate their size relative to other fractional units. In other words, they form ratios.
+
+Consider [this Codepen using fractional units](https://codepen.io/adadev/pen/xWKJdz?editors=1100) on different definitions of `grid-template-columns` on different grid containers. Answer the following questions:
+- Which of these examples mixes fractional units and other units?
+- Which of these creates equally sized columns?
+- Which of these creates three columns?
+
+
+## Grid Item
+
+Now we can manipulate our grid items! For the following examples, we'll be using [this Codepen](https://codepen.io/adadev/pen/yKBqQP?editors=1100).
+
+### Using Start and End
+
+We can define on a grid item which column line number the item starts at using `grid-column-start`:
+
+```css
+.container {
+  display: grid;
+  grid-template: 1fr 1fr / 1fr 1fr;
+}
+
+.item {
+  grid-column-start: 1;
+}
+```
+
+Don't forget that CSS Grid's line numbers do not start at "index zero."
+
```

Comment:
You should say "Remember" instead of "Don't forget" here. Someone once told me that when you say "don't forget x", your brain ignores the "don't" as a stopword, and just keeps the "forget x".